sentry/src/test/java/io/sentry/JsonObjectSerializerTest.kt

```diff
@@ -166,10 +175,107 @@ internal class JsonObjectSerializerTest {
+        val inOrder = inOrder(fixture.writer)
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, map)
+        inOrder.verify(fixture.writer).beginObject()
+        inOrder.verify(fixture.writer).name("one")
+        inOrder.verify(fixture.writer).value("en_US")
+        inOrder.verify(fixture.writer).endObject()
+    }
+
+    @Test
+    fun `serialize locale in list`() {
+        val list = listOf<Locale>(Locale.US, Locale.GERMAN)
+        val inOrder = inOrder(fixture.writer)
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, list)
+        inOrder.verify(fixture.writer).beginArray()
+        inOrder.verify(fixture.writer).value("en_US")
+        inOrder.verify(fixture.writer).value("de")
+        inOrder.verify(fixture.writer).endArray()
+    }
+
+    @Test
+    fun `serialize locale in object`() {
+        val obj = ClassWithLocaleProperty(Locale.US)
+        val inOrder = inOrder(fixture.writer)
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, obj)
+        inOrder.verify(fixture.writer).beginObject()
+        inOrder.verify(fixture.writer).name("localeProperty")
+        inOrder.verify(fixture.writer).value("en_US")
+        inOrder.verify(fixture.writer).endObject()
+    }
+
+    @Test
+    fun `serializing AtomicIntegerArray`() {
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, AtomicIntegerArray(arrayOf(1, 2, 3).toIntArray()))
+        verify(fixture.writer).beginArray()
+        verify(fixture.writer).value(1 as Number)
+        verify(fixture.writer).value(2 as Number)
+        verify(fixture.writer).value(3 as Number)
+        verify(fixture.writer).endArray()
+    }
+
+    @Test
+    fun `serializing AtomicBoolean`() {
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, AtomicBoolean(true))
+        verify(fixture.writer).value(true)
+    }
+
+    @Test
+    fun `serializing URI`() {
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, URI("http://localhost:8081/api/product?id=99"))
+        verify(fixture.writer).value("http://localhost:8081/api/product?id=99")
+    }
+
+    @Test
+    fun `serializing UUID`() {
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, UUID.fromString("828900a5-15dc-413f-8c17-6ef04d74e074"))
+        verify(fixture.writer).value("828900a5-15dc-413f-8c17-6ef04d74e074")
+    }
+
+    @Test
+    fun `serializing Currency`() {
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, Currency.getInstance("USD"))
+        verify(fixture.writer).value("USD")
+    }
+
+    @Test
+    fun `serializing Calendar`() {
+        val calendar = Calendar.getInstance()
+        calendar.set(2022, 0, 1, 11, 59, 58)
+        fixture.getSUT().serialize(fixture.writer, fixture.logger, calendar)
+        verify(fixture.writer).beginObject()
+        verify(fixture.writer).name("year")
+        verify(fixture.writer).value(2022L as java.lang.Long)
```

Comment:
2022L should be enough, shouldn't it?

Reply:
It didn't like that. I assume it's some mockito kotlin / java interop thing.